ipykernel/ipkernel.py

```diff
@@ -60,6 +60,14 @@ def __init__(self, **kwargs):
         self.comm_manager = CommManager(shell=self.shell, parent=self,
                                         kernel=self)
 
+        # BEGIN HARDCODED WIDGETS HACK
+        try:
+            import ipywidgets
+            ipywidgets.handle_kernel(self)
+        except:
+            self.log.warn('ipywidgets package not installed.  Widgets will not be available.')
```

Comment:
since this is on-start, not on-request, I think debug or info level is more appropriate than warn.